source/common/stats/histogram_impl.cc

```diff
@@ -72,12 +74,38 @@ void HistogramStatisticsImpl::refresh(const histogram_t* new_histogram_ptr) {
 
   ASSERT(supportedBuckets().size() == computed_buckets_.size());
   computed_buckets_.clear();
-  const std::vector<double>& supported_buckets = supportedBuckets();
+  ConstSupportedBuckets& supported_buckets = supportedBuckets();
   computed_buckets_.reserve(supported_buckets.size());
   for (const auto bucket : supported_buckets) {
     computed_buckets_.emplace_back(hist_approx_count_below(new_histogram_ptr, bucket));
   }
 }
 
+HistogramSettingsImpl::HistogramSettingsImpl(const envoy::config::metrics::v3::StatsConfig& config)
+    : configs_([&config]() {
+        std::vector<Config> configs;
+        for (const auto& matcher : config.histogram_bucket_settings()) {
+          configs.emplace_back(matcher.match(), ConstSupportedBuckets{matcher.buckets().begin(),
+                                                                      matcher.buckets().end()});
```

Comment:
Is some additional validation required here?

Do the entries in matcher.buckets() need to be sorted?
Can entries include 0.0 or negative numbers?
What if a bucket number appears multiple times?

Reply:
That's a really good point. I'll add some validation now while this change is still fresh, and it won't break anything existing.